spec/handlers/sessions_callback_spec.rb

```diff
@@ -4,237 +4,308 @@
-        )
-
-        expect(result.outputs[:status]).to eq :returning_user
-        expect(user_state.current_user).to eq signed_in_user
-      end
-    end
-
-    context "when auth not linked to a user" do
-      let(:authentication) { FactoryGirl.create(:authentication) }
-
-      it "adds the auth to the signed in user and returns to app" do
-
-        auth_data = {provider: authentication.provider, uid: authentication.uid}
-        result = nil
-        expect{
+  context "not previously signed in" do
+    context "new authorization" do
+      context "with no email or with an email not in the database" do
+        # Identity and authentication already exist,
+        # as they were created during the OAuth request phase
+        let!(:identity)       { FactoryGirl.create(:identity) }
+        let!(:authentication) { FactoryGirl.create(:authentication,
+                                                   user: identity.user,
+                                                   uid: identity.id.to_s,
+                                                   provider: 'identity') }
+
+        it "makes new user and prompts new or returning" do
           result = SessionsCallback.handle(
             user_state: user_state,
-            request: MockOmniauthRequest.new(authentication.provider, authentication.uid, [])
+            request: MockOmniauthRequest.new('identity', identity.user.id, {})
           )
-        }.to change{signed_in_user.authentications.count}.by 1
-        expect(result.outputs[:status]).to eq :returning_user
-        expect(user_state.current_user).to eq signed_in_user
+          
+          expect(result.outputs[:status]).to eq :new_user
+
+          expect(user_state.current_user).not_to be_nil
+          expect(user_state.current_user.person).to be_nil
+          expect(user_state.current_user.is_temp).to be_true
+          
+          linked_authentications = user_state.current_user.authentications
+          expect(linked_authentications.size).to eq 1
+          expect(linked_authentications.first.provider).to eq 'identity'
+          expect(linked_authentications.first.uid).to eq "1"
+        end
       end
-    end
 
-    context "when auth linked to a temp user other than that signed in" do
-      let(:other_temp_user) { FactoryGirl.create(:temp_user) }
-      let(:authentication) { FactoryGirl.create(:authentication, user: other_temp_user) }
-
-      it "transfers temp user auths to signed in user, destroys temp user, returns to app" do
-        result = SessionsCallback.handle(
-          user_state: user_state,
-          request: MockOmniauthRequest.new(authentication.provider, authentication.uid, [])
-        )        
-        expect(authentication.reload.user).to eq signed_in_user
-        expect(result.outputs[:status]).to eq :returning_user
-        expect(User.exists?(other_temp_user.id)).to be_false
+      context "with an email that matches existing emails" do
+        context "for one user" do
```

Comment:
This test case should cover Ross's case. But honestly it's hard to even think about these tests.